sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -1009,35 +1005,4 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog {
       case _ =>
     }
   }
-
-  // Make sure that the `SHOW PARTITIONS` command is allowed for the table
-  private def checkShowPartitions(showPartitions: ShowPartitions): Unit = showPartitions match {
-    case ShowPartitions(rt: ResolvedTable, _, _)
-        if !rt.table.isInstanceOf[SupportsPartitionManagement] =>
-      failAnalysis("SHOW PARTITIONS cannot run for a table which does not support partitioning")
-    case ShowPartitions(ResolvedTable(_, _, partTable: SupportsPartitionManagement, _), _, _)
-        if partTable.partitionSchema().isEmpty =>
```

Comment:
Where is the check now?

Reply:
If a `SupportsPartitionManagement` has empty partition schema, `ResolvePartitionSpec` fails earlier when calling `normalizePartitionSpec`